### DISCUSSION TITLE
Empty array values aren't included when opening the row editor

### DISCUSSION THREAD

**Discussion by probablykasper** [supabase/supabase#16973] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Empty array values aren't included when opening the row editor

## To Reproduce

Edit a record with `[]` like this
<img width="266" alt="image" src="https://github.com/supabase/supabase/assets/11315492/24f02b6a-f615-4657-9c1e-eadbeb034421">

Click Save without making any changes. Result:
<img width="723" alt="image" src="https://github.com/supabase/supabase/assets/11315492/4aee42fb-9566-4318-8f0c-004815790499">

The error actually shows up even if the column is nullable

## Expected behavior

When editing a `[]` cell, it should not change to ``.

## System information

- OS: macOS
- Browser: Brave
- Supabase 1.64.2
- Node.js v18.14.2

**Comment by afrieirham** [comment]
I'm interested to work on this, if anyone could assign this to me that would be great!

**Comment by gregnr** [comment]
Go for it @afrieirham!
